modules/partition-distribution/src/main/java/org/apache/ignite/internal/partitiondistribution/RendezvousDistributionFunction.java

```diff
@@ -284,11 +288,12 @@ public static <T extends Collection<String>> List<T> assignPartitions(
     }
 
     @Override
-    public List<List<String>> assignPartitions(
+    public List<Set<Assignment>> assignPartitions(
             Collection<String> nodes,
             List<List<String>> currentDistribution,
             int partitions,
-            int replicaFactor
+            int replicaFactor,
+            int consensusGroupSize
     ) {
         return assignPartitions(nodes, partitions, replicaFactor, false, null);
```

Comment:
It seems only the first _consensusGroupSize_ assignments have to be marked as a peer and the rest as a learner (isPeer=false).

Reply:
it was done under [IGNITE-24064](https://issues.apache.org/jira/browse/IGNITE-24064)